src/App.tsx

```diff
@@ -1,26 +1,165 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import * as todoService from './api/todos';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { SortBy } from './types/SortBy';
+import { showErrorMesage } from './utils/showErrorMesage';
 
 export const App: React.FC = () => {
+  //#region States
+  //Todos states
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodo, setLoadingTodo] = useState<Todo | null>(null);
+
+  //Service states
+  const [selectedSort, setSelectedSort] = useState<SortBy>(SortBy.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  //Header constans
+  const [todoTitle, setTodoTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  //FooterList
+  const [deletingListId, setDeletingListId] = useState<number[]>([]);
+  //#endregion
+
+  //#region Functions
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleFocusInput = () => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  const sortList = (sort: SortBy) => {
+    switch (sort) {
+      case SortBy.Active:
+        return todos.filter(todo => !todo.completed);
+      case SortBy.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const sortedArray = sortList(selectedSort);
+
+  function addTodos(title: string, completed: boolean, userId: number) {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  const addTodo = (title: string, completed: boolean, userId: number) => {
```